src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,27 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        StringBuilder tempString = new StringBuilder();
+        Set<String> treeSet = new TreeSet<>();
+        String result;
+        String line;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] arrS = line.replaceAll("[^a-zA-Z]", "").toLowerCase().split("");
+                treeSet.addAll(Arrays.asList(arrS));
+            }
+        } catch (IOException e) {
+            throw new FileNotFoundException("Where is the file?" + e);
+        }
+        for (String string : treeSet) {
+            tempString.append(string);
+        }
```

Comment:
can we use counter here of some other check to stop appending when we have reached required size?

Reply:
?

Reply:
done